core/io/resource_loader.cpp

```diff
@@ -160,24 +148,45 @@ void ResourceFormatLoader::get_recognized_extensions(List<String> *p_extensions)
 	}
 }
 
+// Warning: Derived classes must override either `load` or `load_interactive`. The base code
+// here can trigger an infinite recursion otherwise, since `load` calls `load_interactive`
+// vice versa.
+
+Ref<ResourceInteractiveLoader> ResourceFormatLoader::load_interactive(const String &p_path, const String &p_original_path, Error *r_error) {
+	// Warning: See previous note about the risk of infinite recursion.
+	Ref<Resource> res = load(p_path, p_original_path, r_error);
+	if (res.is_null()) {
+		return Ref<ResourceInteractiveLoader>();
+	}
+
+	Ref<ResourceInteractiveLoaderDefault> ril = Ref<ResourceInteractiveLoaderDefault>(memnew(ResourceInteractiveLoaderDefault));
+	ril->resource = res;
+	return ril;
+}
+
 RES ResourceFormatLoader::load(const String &p_path, const String &p_original_path, Error *r_error) {
+	// Check user-defined loader if there's any. Hard fail if it returns an error.
 	if (get_script_instance() && get_script_instance()->has_method("load")) {
 		Variant res = get_script_instance()->call("load", p_path, p_original_path);
 
-		if (res.get_type() == Variant::INT) {
+		if (res.get_type() == Variant::INT) { // Error code, abort.
 			if (r_error) {
 				*r_error = (Error)res.operator int64_t();
 			}
+			const String script_path = get_script().is_valid() ? vformat("'%s'", get_script()->get_path()) : "a user script";
+			ERR_FAIL_V_MSG(RES(),
+					vformat("Failed to load resource '%s' using overridden `load` method in %s. Error code: %d.",
```

Comment:
Functions that return error codes do not print error to console. Its either one or the other.

Reply:
Alright, that's sad because we don't properly use `r_error`, but it's better than a crash in any case. Discussion from Godot chat for reference:
![Screenshot_20210505_154702](https://user-images.githubusercontent.com/4701338/117151281-37b38900-adb9-11eb-948f-a42b66599b10.png)